File: src/main/scala-graphql/io/github/graphql/scala/model/UpdatePullRequestReviewCommentInputTO.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:37+0800"
)
class UpdatePullRequestReviewCommentInputTO(
    @javax.validation.constraints.NotNull
    body: String,
    clientMutationId: String,
    @javax.validation.constraints.NotNull
    pullRequestReviewCommentId: String
) {

    override def toString(): String = {
        Seq(
            if (body != null) "body: " + GraphQLRequestSerializer.getEntry(body) else "",
            if (clientMutationId != null) "clientMutationId: " + GraphQLRequestSerializer.getEntry(clientMutationId) else "",
            if (pullRequestReviewCommentId != null) "pullRequestReviewCommentId: " + GraphQLRequestSerializer.getEntry(pullRequestReviewCommentId) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[UpdatePullRequestReviewCommentInputTO]
        Objects.equals(body, that.body) &&
        Objects.equals(clientMutationId, that.clientMutationId) &&
        Objects.equals(pullRequestReviewCommentId, that.pullRequestReviewCommentId)
    }

    override def hashCode(): Int = {
        Objects.hash(body, clientMutationId, pullRequestReviewCommentId)
    }
}

object UpdatePullRequestReviewCommentInputTO {

    def builder(): UpdatePullRequestReviewCommentInputTO.Builder = new Builder()

    class Builder {

        private var body: String = _
        private var clientMutationId: String = _
        private var pullRequestReviewCommentId: String = _

        def setBody(body: String): Builder = {
            this.body = body
            this
        }

        def setClientMutationId(clientMutationId: String): Builder = {
            this.clientMutationId = clientMutationId
            this
        }

        def setPullRequestReviewCommentId(pullRequestReviewCommentId: String): Builder = {
            this.pullRequestReviewCommentId = pullRequestReviewCommentId
            this
        }

        def build(): UpdatePullRequestReviewCommentInputTO = new UpdatePullRequestReviewCommentInputTO(body, clientMutationId, pullRequestReviewCommentId)

    }
}

File: src/main/java-graphql/io/github/graphql/java/model/UpdatePullRequestReviewCommentInputTO.java
package io.github.graphql.java.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.Objects;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public class UpdatePullRequestReviewCommentInputTO implements java.io.Serializable {

    @javax.validation.constraints.NotNull
    private String body;
    private String clientMutationId;
    @javax.validation.constraints.NotNull
    private String pullRequestReviewCommentId;

    public UpdatePullRequestReviewCommentInputTO() {
    }

    public UpdatePullRequestReviewCommentInputTO(String body, String clientMutationId, String pullRequestReviewCommentId) {
        this.body = body;
        this.clientMutationId = clientMutationId;
        this.pullRequestReviewCommentId = pullRequestReviewCommentId;
    }

    public String getBody() {
        return body;
    }

    public String getClientMutationId() {
        return clientMutationId;
    }

    public String getPullRequestReviewCommentId() {
        return pullRequestReviewCommentId;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        final UpdatePullRequestReviewCommentInputTO that = (UpdatePullRequestReviewCommentInputTO) obj;
        return Objects.equals(body, that.body)
            && Objects.equals(clientMutationId, that.clientMutationId)
            && Objects.equals(pullRequestReviewCommentId, that.pullRequestReviewCommentId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(body, clientMutationId, pullRequestReviewCommentId);
    }

    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (body != null) {
            joiner.add("body: " + GraphQLRequestSerializer.getEntry(body));
        }
        if (clientMutationId != null) {
            joiner.add("clientMutationId: " + GraphQLRequestSerializer.getEntry(clientMutationId));
        }
        if (pullRequestReviewCommentId != null) {
            joiner.add("pullRequestReviewCommentId: " + GraphQLRequestSerializer.getEntry(pullRequestReviewCommentId));
        }
        return joiner.toString();
    }

    public static UpdatePullRequestReviewCommentInputTO.Builder builder() {
        return new UpdatePullRequestReviewCommentInputTO.Builder();
    }

    public static class Builder {

        private String body;
        private String clientMutationId;
        private String pullRequestReviewCommentId;

        public Builder() {
        }

        public Builder setBody(String body) {
            this.body = body;
            return this;
        }

        public Builder setClientMutationId(String clientMutationId) {
            this.clientMutationId = clientMutationId;
            return this;
        }

        public Builder setPullRequestReviewCommentId(String pullRequestReviewCommentId) {
            this.pullRequestReviewCommentId = pullRequestReviewCommentId;
            return this;
        }


        public UpdatePullRequestReviewCommentInputTO build() {
            return new UpdatePullRequestReviewCommentInputTO(body, clientMutationId, pullRequestReviewCommentId);
        }

    }
}
